### DISCUSSION TITLE
Why is cargo failing to build Bevy due to unstable features? I thought Bevy supported stable Rust versions?

### DISCUSSION THREAD

**Discussion by montreal91** [bevyengine/bevy#6684] [Q&A] [answered]
Hello everyone, Bevy is Awesome!

I'm trying to make a hello world with bevy 0.9 and all of a sudden, I can't. 

My Cargo.toml is dead simple 

```toml
[package]
name = "bevy0-9"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
bevy = "0.9"
```
When I enter `cargo build` command, I get this error.

```
   Compiling bevy_derive v0.9.0
error[E0658]: `let...else` statements are unstable
   --> ~\.cargo\registry\src\github.com-1ecc6299db9ec823\bevy_reflect_derive-0.9.0\src\container_attributes.rs:140:21
    |
140 | /                     let Some(segment) = path.segments.iter().next() else {
141 | |                         continue;
142 | |                     };
    | |______________________^
    |
    = note: see issue #87335 <https://github.com/rust-lang/rust/issues/87335> for more information
```
Bevy 0.8.1 works well, so I don't know what to think. Looks like some kind of bug, but I'm not sure.

**Comment by hymm** [comment, marked as answer]
You need to update rust to 1.65 or newer. Run `rustup update`.

**Reply by montreal91** [reply]
Thanks a lot, that helped :)

**Reply by MrGVSV** [reply]
Note: this may need to be followed by a `cargo clean`

**Reply by dakitten2358** [reply]
I was also bitten by this, and the error message isn't super obvious that the compiler needs updating.  Would adding rust-version to Bevy's cargo.toml help?

https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field

**Reply by zicklag** [reply]
Good idea! I'll open a PR, unless you want to do the honors?

You can do it right from the GitHub UI, if you don't want to clone the repo or install anything.

**Reply by mockersf** [reply]
we've been discussing that field a few times on issues/prs/discord... if it's added, there need to be a job in CI that validates that the version specified is still valid

**Reply by zicklag** [reply]
Ah, I see. That get's more complicated, then.
